kroxylicious/src/test/java/io/kroxylicious/proxy/internal/FilterHandlerTest.java

```diff
@@ -698,9 +777,9 @@ void sendAcklessProduceRequest() {
     @Test
     void sendRequestTimeout() {
         var oobRequestBody = new FetchRequestData();
-        var snoopedOobRequestResponseStage = new AtomicReference<CompletionStage<ApiMessage>>();
+        var snoopedOobRequestResponseStage = new AtomicReference<CompletionStage<ResponseHeaderAndApiMessage<FetchResponseData>>>();
         ApiVersionsRequestFilter filter = (apiVersion, header, request, context) -> {
-            snoopedOobRequestResponseStage.set(context.sendRequest((short) 3, oobRequestBody));
+            snoopedOobRequestResponseStage.set(context.sendRequest(new RequestHeaderData(), oobRequestBody));
```

Comment:
should the new header preserve the old API version `3`?

Reply:
From this perspective of this test, version 0 of FetchRequest is good enough.  I don't think 3 was ever actually significant to this test. `new RequestHeaderData()` will default  the version to 0.